packages/table-core/src/features/Sorting.ts

```diff
@@ -191,6 +192,9 @@ export const Sorting: TableFeature = {
         //   return
         // }
 
+        // this needs to be outside of instance.setSorting to be in sync with rerender
+        const nextSortingOrder = column.getNextSortingOrder()
```

Comment:
@tannerlinsley I first put this line inside `instance.setSorting` but then the react sorting example broke when `sortAction` became `toggle`. It wasn't clear to me exactly why but I assume some state not being in sync on rerender.